src/coreclr/inc/corinfo.h

```diff
@@ -2893,9 +2893,15 @@ class ICorStaticInfo
             CORINFO_METHOD_HANDLE hMethod
             ) = 0;
 
-    // this function is for debugging only.  It returns the method name
-    // and if 'moduleName' is non-null, it sets it to something that will
-    // says which method (a class name, or a module name)
+    // This function returns the method name and if 'moduleName' is non-null,
+    // it sets it to something that will says what contains the method (a class
+    // name, or a module name). Note that the moduleName parameter is for
```

Comment:
grammar
```suggestion
    // it sets it to something that contains the method (a class
    // name, or a module name). Note that the moduleName parameter is for
```

Also, what does it mean "for diagnostics only"?

Reply:
In the EE, `moduleName` is set to basically "what it can get its hands on": in debug builds it has a debug-only buffer available that it uses to create a namespace qualified class name. But in release builds it does not have any buffer available, so it just returns a pointer to the class name only. So using `moduleName` from the JIT introduces different behavior depending on whether coreclr.dll is debug or release.

This PR stops using this parameter and instead uses `getClassNameFromMetadata` to create the namespace qualified names inside the JIT. We still use `getMethodName`, but only the returned method name, which seems to not have this "debug-only/diagnostic-only" flavor to it, so I have updated the documentation to try to make this more clear.